src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -890,11 +889,12 @@ SequenceNumber_t StatefulWriter::next_sequence_number() const
 void StatefulWriter::send_heartbeat_to_nts(ReaderProxy& remoteReaderProxy, bool final)
 {
     std::vector<GUID_t> tmp_guids(1, remoteReaderProxy.m_att.guid);
-    const LocatorList_t& locators = remoteReaderProxy.m_att.endpoint.remoteLocatorList;
+    const LocatorList_t remote_locators_shrinked{mp_RTPSParticipant->network_factory().ShrinkLocatorLists(
+            {remoteReaderProxy.m_att.endpoint.remoteLocatorList})};
```

Comment:
If I remeber correctly, remoteLocatorList is shrinked inside `matched_reader_add`, so there is no need to shrink it here

Reply:
Decided by the team to keep in this way.